OpenRA.Mods.Common/Pathfinder/HierarchicalPathFinder.cs

```diff
@@ -146,12 +146,21 @@ public GridInfo(CPos?[] singleAbstractCellForLayer, Dictionary<CPos, CPos> local
 			/// <summary>
 			/// Maps a local cell to a abstract node in the graph.
 			/// Returns null when the local cell is unreachable.
+			/// Pass a null <paramref name="hpf"/> to skip cost checks if the caller already checked.
 			/// </summary>
-			public CPos? AbstractCellForLocalCell(CPos localCell)
+			public CPos? AbstractCellForLocalCell(CPos localCell, HierarchicalPathFinder hpf)
 			{
 				var abstractCell = singleAbstractCellForLayer[localCell.Layer];
 				if (abstractCell != null)
+				{
+					// All reachable cells in the grid are joined together so only a single abstract cell was needed,
+					// but there may be unreachable cells in the grid which we must exclude.
+					if (hpf != null && !hpf.CellIsAccessible(localCell))
+						return null;
 					return abstractCell;
```

Comment:
```suggestion
						return null;

					return abstractCell;
```

Reply:
Actually, never mind. Cramping everything together is the code style of this file.